crates/walrus-sdk/src/client/communication/node.rs

```diff
@@ -362,11 +363,28 @@ impl NodeWriteCommunication {
         &self,
         metadata: &VerifiedBlobMetadataWithId,
         pairs: impl IntoIterator<Item = &SliverPair>,
+    ) -> NodeResult<(), StoreError> {
+        self.store_metadata_and_pairs_without_confirmation_with_intent(
+            metadata,
+            pairs,
+            UploadIntent::Immediate,
```

Comment:
suggestion: instead of hard code UploadIntent::Immediate, should we add this as an input to store_metadata_and_pairs_without_confirmation to be a bit more flexible?